CMakeLists.txt

```diff
@@ -1,332 +1,374 @@
-        endif()
-    else()
-        message(FATAL_ERROR "Git not detected: cannot obtain submodule at \"${PATH}\" (but how is this even working?)")
-    endif()
-endfunction()
-
-set(PROJ_ROOT_INCLUDE_DIR ./include)
-set(PROJ_ROOT_MODELS_DIR ./models)
-
+# -----------------------------------------------------------------------------
 # Find the Boost library and configure usage
 set(Boost_USE_STATIC_LIBS OFF)
 set(Boost_USE_MULTITHREADED ON)
 set(Boost_USE_STATIC_RUNTIME OFF)
-
 if(NOT DEFINED BOOST_ROOT AND NOT DEFINED ENV{BOOST_ROOT})
     # Look for version-specific Boost directory if available (known from Github Actions VM docs)
     if(DEFINED ENV{BOOST_ROOT_1_72_0})
         set(BOOST_ROOT $ENV{BOOST_ROOT_1_72_0})
-        message(STATUS "Using value of \$ENV{BOOST_ROOT_1_72_0} to set BOOST_ROOT=$ENV{BOOST_ROOT_1_72_0}")
-    else()
-        message(STATUS "No BOOST_ROOT set")
     endif()
 endif()
-
 find_package(Boost 1.72.0 REQUIRED)
 
-if(MPI_ACTIVE)
-    find_package(MPI)
-    add_compile_definitions(NGEN_MPI_ACTIVE)
-endif()
-
-find_package(SQLite3)
-if(SQLite3_FOUND)
-    set(NGEN_WITH_SQLITE3 ON)
+# -----------------------------------------------------------------------------
+if(NGEN_WITH_SQLITE)
+    find_package(SQLite3 REQUIRED)
     add_compile_definitions(NGEN_WITH_SQLITE3)
-else()
-    set(NGEN_WITH_SQLITE3 OFF)
-endif()
-message(INFO " SQLite3 support is ${NGEN_WITH_SQLITE3}")
-
-# UDUNITS
-# Since UDUNITS is currently not really optional (yet) let's make it default to on...
-if (NOT (DEFINED UDUNITS_ACTIVE))
-    message(INFO " UDUNITS_ACTIVE not defined")
-    if (DEFINED ENV{UDUNITS_ACTIVE})
-        message(INFO " set as $ENV{UDUNITS_ACTIVE}")
-        set(UDUNITS_ACTIVE $ENV{UDUNITS_ACTIVE})
-    else()
-        set(UDUNITS_ACTIVE true)
-    endif()
 endif()
 
-if(UDUNITS_ACTIVE)
+# -----------------------------------------------------------------------------
+if(NGEN_WITH_UDUNITS)
     find_package(UDUNITS2 REQUIRED)
-    if(UDUNITS2_LIBRARY)
-        #add_library(libudunits2 SHARED IMPORTED)
-        target_include_directories(libudunits2 INTERFACE ${UDUNITS2_INCLUDE})
-        #include_directories(${UDUNITS2_INCLUDE})
-        message("INFO Using UDUNITS2 at ${UDUNITS2_LIBRARY} and ${UDUNITS2_INCLUDE}")
-    endif()
-    #add_compile_definitions("NGEN_UDUNITS2_XML_PATH=\"${UDUNITS_INSTALL_DIR}/share/udunits/udunits2.xml\"")
+    add_compile_definitions(NGEN_WITH_UDUNITS)
+
+    # Create UDUNITS2 target
+    # Note: GLOBAL is required here in order to extend the scope of the target.
+    #       see: https://stackoverflow.com/a/46491758/6891484
+    add_library(libudunits2 SHARED IMPORTED GLOBAL)
+    set_target_properties(libudunits2 PROPERTIES IMPORTED_LOCATION "${UDUNITS2_LIBRARY}")
+    target_include_directories(libudunits2 INTERFACE "${UDUNITS2_INCLUDE}")
+        
+    get_target_property(_SUMMARY_UDUNITS_INCLUDE libudunits2 INTERFACE_INCLUDE_DIRECTORIES)
```

Comment:
If there's any other changes to be made, it would be kinda nice if this were all encapsulated in `FindUDUNITS2.cmake`. It would also make that more share-able with other projects as a result.

Reply:
Yeah, I think part of this was initially in there, but I moved it out for some reason (couldn't tell you why haha). I see no problems with moving this and the NetCDF target creation into their respective modules. I'll make a short one for the iso_c_bmi_fortran target too, since then it can just be a `find_package` call.